plugins/stack-overflow-backend/src/search/StackOverflowQuestionsCollatorFactory.ts

```diff
@@ -93,12 +100,16 @@ export class StackOverflowQuestionsCollatorFactory
         `No stackoverflow.baseUrl configured in your app-config.yaml`,
       );
     }
+
     const params = qs.stringify(this.requestParams, {
       arrayFormat: 'comma',
       addQueryPrefix: true,
     });
 
-    const res = await fetch(`${this.baseUrl}/questions${params}`);
+    const apiKey = this.config.getOptionalString('stackoverflow.apiKey');
+    const apiKeyParam = apiKey ? `${params ? '&' : '?'}key=${apiKey}` : '';
+
+    const res = await fetch(`${this.baseUrl}/questions${params}${apiKeyParam}`);
```

Comment:
Thanks for contributing @mmulligan03 🎉 
It looks like we can already pass a key in the request parameters, something like this:
```ts
indexBuilder.addCollator({
  schedule: everyHourSchedule,
  factory: StackOverflowQuestionsCollatorFactory.fromConfig(env.config, {
    logger: env.logger,
    requestParams: {
      key: 'YOUR KEY',
      pagesize: 100,
    },
  }),
});
```
Do you know (or maybe @emmaindal knows) what the difference is in using `requestParams` directly?

Reply:
yes @camilaibs please look at the comment in the issues https://github.com/backstage/backstage/issues/11531 for the reason we decided on not passing the key through request params.

Reply:
Thank you for pointing that out!